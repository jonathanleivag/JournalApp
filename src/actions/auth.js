import { types } from "../types/types";
import { firebase, googleAuthProvider } from "../firebase/firebaseConfig";
import { finishLoading, startLoading } from "./ui";
import Swal from "sweetalert2";

export const startRegister = (name, email, password) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((error) => {
        Swal.fire("Error", error.message, "error");
      });
  };
};

export const startLogin = (email, password) => {
  return async (dispatch) => {
    dispatch(startLoading());
    try {
      const {
        user: { uid, displayName },
      } = await firebase.auth().signInWithEmailAndPassword(email, password);
      dispatch(login(uid, displayName));
      dispatch(finishLoading());
    } catch (error) {
      dispatch(finishLoading());
      Swal.fire("Error", error.message, "error");
    }
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user: { uid, displayName } }) => {
        dispatch(login(uid, displayName));
      })
      .catch((error) => Swal.fire("Error", error.message, "error"));
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: { uid, displayName },
});

export const startLogout = () => {
  return async (dispatch) => {
    try {
      await firebase.auth().signOut();
      dispatch(logout());
    } catch (error) {
      Swal.fire("Error", error.message, "error");
    }
  };
};

export const logout = () => ({ type: types.logout });
