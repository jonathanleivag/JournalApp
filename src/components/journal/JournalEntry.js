import React from "react";

export const JournalEntry = () => {
  const style = {
    backgroundSize: "cover",
    backgroundImage:
      "url(https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__340.jpg)",
  };

  return (
    <div className="journal__entry pointer">
      <div className="journal__entry-picture" style={style}></div>
      <div className="journal__entry-body">
        <p className="journal__entry-title">Lorem Ipsum</p>
        <p className="journal__entry-content">
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry.
        </p>
      </div>

      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>26</h4>
      </div>
    </div>
  );
};
