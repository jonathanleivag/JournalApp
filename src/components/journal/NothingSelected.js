import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const NothingSelected = () => {
  return (
    <div className="nothing__main-content">
      <p>
        Select something <br /> pr create an entry
      </p>
      <FontAwesomeIcon icon={["far", "star"]} size="4x" className="mt-5" />
    </div>
  );
};
