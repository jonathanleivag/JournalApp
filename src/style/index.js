import "./style.js";
import "./style.scss";
