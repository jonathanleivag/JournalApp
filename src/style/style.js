import "./fontawesome/fontawesome";
import "animate.css/animate.min.css";
