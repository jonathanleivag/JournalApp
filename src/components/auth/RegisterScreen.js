import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";
import validator from "validator";
import { useDispatch, useSelector } from "react-redux";
import { setError, removeError } from "../../actions/ui";
import { startRegister } from "../../actions/auth";

export const RegisterScreen = () => {
  const [values, handleInputChange] = useForm({
    name: "",
    email: "",
    password: "",
    password2: "",
  });

  const { name, email, password, password2 } = values;
  const dispatch = useDispatch();
  const { msgError } = useSelector((state) => state.ui);

  const handlerRegister = (e) => {
    e.preventDefault();
    if (isFormValid()) {
      dispatch(startRegister(name, email, password));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError("Name is required"));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError("Email is not valid"));
      return false;
    } else if (password !== password2 || password.length < 5) {
      dispatch(setError("Password shoult be at least 6 characters and match"));
      return false;
    }
    dispatch(removeError());
    return true;
  };

  return (
    <Fragment>
      <h1 className="auth__title">Register Screen</h1>

      <form autoComplete="off" onSubmit={handlerRegister}>
        {msgError !== null && (
          <div className="auth__alert-error">{msgError}</div>
        )}

        <input
          type="text"
          placeholder="Name"
          name="name"
          className="auth__input"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />

        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
          autoComplete="off"
          value={password}
          onChange={handleInputChange}
        />

        <input
          type="password"
          placeholder="Confirm"
          name="password2"
          className="auth__input"
          autoComplete="off"
          value={password2}
          onChange={handleInputChange}
        />

        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>

        <Link to="/auth/login" className="link">
          Already registered ?
        </Link>
      </form>
    </Fragment>
  );
};
