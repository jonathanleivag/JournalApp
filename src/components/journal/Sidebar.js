import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { JournalEntries } from "./JournalEntries";
import { useDispatch } from "react-redux";
import { startLogout } from "../../actions/auth";

export const Sidebar = () => {
  const dispatch = useDispatch();

  const handlerLogout = () => {
    dispatch(startLogout());
  };

  return (
    <aside className="journal__sidebar">
      <div className="journal__sidebar-navbar">
        <h3 className="mt-5">
          <FontAwesomeIcon icon={["far", "moon"]} />
          <span className="space">Jonathan</span>
        </h3>
        <button className="btn" onClick={handlerLogout}>
          Logout
        </button>
      </div>

      <div className="journal__new-entry">
        <FontAwesomeIcon icon={["far", "calendar-plus"]} size="5x" />
        <p className="mt-5">New entry</p>
      </div>
      <JournalEntries />
    </aside>
  );
};
