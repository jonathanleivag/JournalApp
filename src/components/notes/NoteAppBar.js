import React, { useRef } from "react";
import { useDispatch, useSelector } from "react-redux";
import { startSaveNote, startUploading } from "../../actions/notes";

// react-journal

// https://api.cloudinary.com/v1_1/dmjxpnr5g

export const NoteAppBar = () => {
  const { active } = useSelector((state) => state.notes);
  const dispatch = useDispatch();
  const handlerSave = () => {
    dispatch(startSaveNote(active));
  };

  const style = { display: "none" };

  const ref = useRef();

  const handlerPictureClick = () => ref.current.click();

  const handlerChangePicture = () => {
    const file = ref.current.files[0];
    if (file) dispatch(startUploading(file));
  };

  return (
    <div className="notes__appbar">
      <span>28 de Agosto 2020</span>

      <input
        type="file"
        id="file"
        style={style}
        onChange={handlerChangePicture}
        ref={ref}
        accept="image/*"
      />

      <div>
        <button className="btn" onClick={handlerPictureClick}>
          Picture
        </button>
        <button className="btn" onClick={handlerSave}>
          Save
        </button>
      </div>
    </div>
  );
};
