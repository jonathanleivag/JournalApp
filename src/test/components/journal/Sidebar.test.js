import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { Sidebar } from "../../../components/journal/Sidebar";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import "../../../style/index";
import { startLogout } from "../../../actions/auth";
import { startNewNotes } from "../../../actions/notes";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock("../../../actions/auth", () => ({
  startLogout: jest.fn(),
}));

jest.mock('../../../actions/notes', () => ({
    startNewNotes: jest.fn(),
}));


const initState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: {
      id: "ABC",
    },
    notes: [],
  },
};
let store = mockStore(initState);
store.dispatch = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <Sidebar />
  </Provider>
);

describe("Pruebas en <Sidebar />", () => {
  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de llamar el startLogout", () => {
    wrapper.find("button").prop("onClick")();
    expect(startLogout).toHaveBeenCalled();
  });

  test('debe de llamar el startNewNote', () => {
    wrapper.find('.journal__new-entry').prop('onClick')();
    expect( startNewNotes ).toHaveBeenCalled();
  })

});
