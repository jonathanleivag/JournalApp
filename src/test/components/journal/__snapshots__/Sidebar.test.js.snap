// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Sidebar /> debe de mostrarse correctamente 1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-moon fa-w-16 "
        data-icon="moon"
        data-prefix="far"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.135 512c78.756 0 150.982-35.804 198.844-94.775 28.27-34.831-2.558-85.722-46.249-77.401-82.348 15.683-158.272-47.268-158.272-130.792 0-48.424 26.06-92.292 67.434-115.836 38.745-22.05 28.999-80.788-15.022-88.919A257.936 257.936 0 0 0 279.135 0c-141.36 0-256 114.575-256 256 0 141.36 114.576 256 256 256zm0-464c12.985 0 25.689 1.201 38.016 3.478-54.76 31.163-91.693 90.042-91.693 157.554 0 113.848 103.641 199.2 215.252 177.944C402.574 433.964 344.366 464 279.135 464c-114.875 0-208-93.125-208-208s93.125-208 208-208z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span
        className="space"
      >
         
         
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-calendar-plus fa-w-14 fa-5x "
      data-icon="calendar-plus"
      data-prefix="far"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M336 292v24c0 6.6-5.4 12-12 12h-76v76c0 6.6-5.4 12-12 12h-24c-6.6 0-12-5.4-12-12v-76h-76c-6.6 0-12-5.4-12-12v-24c0-6.6 5.4-12 12-12h76v-76c0-6.6 5.4-12 12-12h24c6.6 0 12 5.4 12 12v76h76c6.6 0 12 5.4 12 12zm112-180v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <p
      className="mt-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal__entries"
  />
</aside>
`;
